### PR TITLE
Refactored network layer to pull shared logic into one place - means we can perform the request to get weather info in multiple places 

### PR DESCRIPTION
**Pull request by jsanderson44**

### CODE REVIEW

File: TideApp/TideApp/Domain/Network/Requests/GetWeatherInformationRequest.swift [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+//  GetWeatherInformationRequest.swift
+//  TideApp
+//
+//  Created by John Sanderson on 19/05/2021.
+//
+
+import CoreLocation
+import Combine
+
+struct GetWeatherInformationRequest: RequestType {
```
Comment (jsanderson44) [inline TideApp/TideApp/Domain/Network/Requests/GetWeatherInformationRequest.swift:11]: The idea is that this "Request" can be made in multiple locations and it hides all the logic around fetching from the API and mapping to a `WeatherInfo` object etc.

File: TideApp/TideApp/Domain/Network/NetworkManager.swift [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import Foundation
+import Combine
+
+protocol NetworkManagerType {
+
+  func fetch<R: ResourceType>(_ resource: R) -> AnyPublisher<R.Output, Error>
+}
+
+final class NetworkManager: NetworkManagerType {
```
Comment (jsanderson44) [inline TideApp/TideApp/Domain/Network/NetworkManager.swift:16]: This is just a slightly more generic network layer that could easily be used to make other requests

Comment (MarcoGuerrieriTAB) [reply TideApp/TideApp/Domain/Network/NetworkManager.swift:16]: Yeah thats was needed indeed! I would also think at some point to move the general network generic layer inside a module

File: TideApp/TideApp/Domain/Network/Requests/GetWeatherInformationRequest.swift (hunk 1 above)
Comment (MarcoGuerrieriTAB) [reply TideApp/TideApp/Domain/Network/Requests/GetWeatherInformationRequest.swift:11]: 👏

**Review by MarcoGuerrieriTAB — APPROVED** [review]
Indeed a really good refactor

**Review by SophieClarkKC — APPROVED** [review]
👏

**Pull request merged**